protocol/src/main/java/org/jboss/as/protocol/ProtocolConnectionManager.java

```diff
@@ -122,6 +122,16 @@ public void shutdown() {
     }
 
     /**
+     * Waits until the connection is closed.
+     */
+    public void awaitConnectionClosed() {
+        final Connection connection = this.connection;
+        if (connection != null) {
+            connection.awaitClosedUninterruptibly();
```

Comment:
Same comment as for https://github.com/wildfly/wildfly-core/pull/2353/files#diff-8a7f63d9e35642093e986ee782c1b369R110